nevergrad/functions/games/board.py

```diff
@@ -0,0 +1,185 @@
+import itertools
+from collections import namedtuple
+# Credits: http://www.moderndescartes.com/essays/implementing_go/
+N = 7
+NN = N ** 2
+WHITE, BLACK, EMPTY = 'O', 'X', '.'
+
+def swap_colors(color):
+    if color == BLACK:
+        return WHITE
+    elif color == WHITE:
+        return BLACK
+    else:
+        return color
+
+EMPTY_BOARD = EMPTY * NN
+
+def flatten(c):
+    return N * c[0] + c[1]
+
+# Convention: coords that have been flattened have a "f" prefix
+def unflatten(fc):
+    return divmod(fc, N)
+
+def is_on_board(c):
+    return c[0] % N == c[0] and c[1] % N == c[1]
+
+def get_valid_neighbors(fc):
+    x, y = unflatten(fc)
+    possible_neighbors = ((x+1, y), (x-1, y), (x, y+1), (x, y-1))
+    return [flatten(n) for n in possible_neighbors if is_on_board(n)]
+
+# Neighbors are indexed by flat coordinates
+NEIGHBORS = [get_valid_neighbors(fc) for fc in range(NN)]
+
+def find_reached(board, fc):
+    color = board[fc]
+    chain = set([fc])
+    reached = set()
+    frontier = [fc]
+    while frontier:
+        current_fc = frontier.pop()
+        chain.add(current_fc)
+        for fn in NEIGHBORS[current_fc]:
+            if board[fn] == color and not fn in chain:
+                frontier.append(fn)
+            elif board[fn] != color:
+                reached.add(fn)
+    return chain, reached
+
+class IllegalMove(Exception): pass
+
+def place_stone(color, board, fc):
+    return board[:fc] + color + board[fc+1:]
+
+def bulk_place_stones(color, board, stones):
+    byteboard = bytearray(board, encoding='ascii') # create mutable version of board
+    color = ord(color)
+    for fstone in stones:
+        byteboard[fstone] = color
+    return byteboard.decode('ascii') # and cast back to string when done
+
+def maybe_capture_stones(board, fc):
+    chain, reached = find_reached(board, fc)
+    if not any(board[fr] == EMPTY for fr in reached):
+        board = bulk_place_stones(EMPTY, board, chain)
+        return board, chain
+    else:
+        return board, []
+
+def play_move_incomplete(board, fc, color):
+    if board[fc] != EMPTY:
+        raise IllegalMove
+    board = place_stone(color, board, fc)
+
+    opp_color = swap_colors(color)
+    opp_stones = []
+    my_stones = []
+    for fn in NEIGHBORS[fc]:
+        if board[fn] == color:
+            my_stones.append(fn)
+        elif board[fn] == opp_color:
+            opp_stones.append(fn)
+
+    for fs in opp_stones:
+        board, _ = maybe_capture_stones(board, fs)
+
+    for fs in my_stones:
+        board, _ = maybe_capture_stones(board, fs)
+
+    return board
+
+def is_koish(board, fc):
+    'Check if fc is surrounded on all sides by 1 color, and return that color'
+    if board[fc] != EMPTY: return None
+    neighbor_colors = {board[fn] for fn in NEIGHBORS[fc]}
+    if len(neighbor_colors) == 1 and not EMPTY in neighbor_colors:
+        return list(neighbor_colors)[0]
+    else:
+        return None
+
+class Position(namedtuple('Position', ['board', 'ko'])):
```

Comment:
Does this work??? I  don't think it's supposed to (your deriving from a structure which has the same name)